Scripts/ModelViewport.js

```diff
@@ -7,132 +7,58 @@
-
-        this.mouseDown = false;
-        this.lastMouseX = null;
-        this.lastMouseY = null;
-
-        document.getElementById(canvasName).onmousedown = (event) => {
-            console.log("detected!\n");
-            this.mouseDown = true;
-            this.lastMouseX = event.clientX;
-            this.lastMouseY = event.clientY;
-        };
-
-        document.getElementById(canvasName).onmouseup = (event) => {
-            this.mouseDown = false;
-        };
-
-        document.getElementById(canvasName).onmousemove = (event) => {
-            if (!this.mouseDown) {
-                return;
-            }
-
-            var newX = event.clientX;
-            var newY = event.clientY;
-
-            var deltaY = newY - this.lastMouseY;
-            var deltaX = newX - this.lastMouseX;
-            if (Math.abs(deltaX) > Math.abs(deltaY)) this.cameraXRotation += 0.01*deltaX;
-            else this.cameraYRotation += 0.01*deltaY;
-            console.log(this.cameraXRotation);
-
-            this.lastMouseX = newX;
-            this.lastMouseY = newY;
-        };
-    }
-
-    setupWebGL2() {
-        this.gl = init_gl_context(this.canvas);
-        this.gl.clearColor(0, 0, 0, 1);
-        this.gl.enable(this.gl.DEPTH_TEST);
-        this.gl.viewportWidth = this.canvas.width;
-        this.gl.viewportHeight = this.canvas.height;
-        this.gl.viewport(0, 0, this.canvas.width, this.canvas.height);
-    }
-
-    //getShader(gl, id) {
-        //var shaderScript = document.getElementById(id);
-        //if (!shaderScript){
-            //return null;
-        //}
-
-        //var str = "";
-        //var k = shaderScript.firstChild;
-        //while (k){
-            //if (k.nodeType == 3){
-                //str += k.textContent;
-            //}
-            //k = k.nextSibling;
-        //}
-
-        //var shader;
-        //if (shaderScript.type == "x-shader/x-fragment"){
-            //shader = gl.createShader(gl.FRAGMENT_SHADER);
-        //} else if (shaderScript.type == "x-shader/x-vertex"){
-            //shader = gl.createShader(gl.VERTEX_SHADER);
-        //} else{
-            //return null;
-        //}
-
-        //gl.shaderSource(shader, str);
-        //gl.compileShader(shader);
-
-        //if (!gl.getShaderParameter(shader, gl.COMPILE_STATUS)){
-            //alert(gl.getShaderInfoLog(shader));
-            //return null;
-        //}
-
-        //return shader;
-    //}
-
-    initShaders() {
-        var gl = this.gl;
-        //var fragmentShader = this.getShader(gl, "shader-fs");
-        //var vertexShader = this.getShader(gl, "shader-vs");
-
-        //this.shaderProgram = gl.createProgram();
-        //gl.attachShader(this.shaderProgram, vertexShader);
-        //gl.attachShader(this.shaderProgram, fragmentShader);
-        //gl.linkProgram(this.shaderProgram);
-
-        //if (!gl.getProgramParameter(this.shaderProgram, gl.LINK_STATUS)){
+export default function ModelViewport(spec) {
```

Comment:
I personally am not a fan of this. ModelViewport should be an object, not a function. Code should be self commenting, and this feels misleading. 

I understand that javascript classes are a bit lacking though. Like, having to use "this" all the time sucks.

Could you put a block comment above functions that we should be pretending are classes/objects?

Will this effect callbacks? For example, "function() {}" vs "() => {}" syntax might suddenly change what the keyword "this" refers to. If I wanted to write a callback, I sometimes pass the keyword "this" as a parameter. Now I don't think I can do that...